server/catgenome/src/main/java/com/epam/catgenome/dao/index/indexer/BigVcfFeatureIndexBuilder.java

```diff
@@ -127,20 +137,38 @@ public BigVcfFeatureIndexBuilder(VcfFilterInfo filterInfo, VCFHeader vcfHeader,
             List<OrganismType> organismTypes = new ArrayList<>();
             for (int i = 0; i < context.getAlternateAlleles().size(); i++) {
                 Variation variation = VcfFileReader.createVariation(context, getVcfHeader(), i);
-                organismTypes.add(variation.getGenotypeData().getOrganismType());
+                organismTypes.addAll(variation.getGenotypeData().values().stream()
+                        .map(GenotypeData::getOrganismType).collect(Collectors.toList()));
             }
 
             if (!organismTypes.isEmpty() && organismTypes.stream()
-                    .anyMatch(type -> type.equals(OrganismType.NO_VARIATION))) {
+                    .allMatch(type -> type.equals(OrganismType.NO_VARIATION))) {
                 return;
             }
 
-            VcfIndexEntry indexEntry = build(masterEntry, geneFiles, chromosome);
-            Document document = creator.buildDocument(indexEntry, vcfFile.getId());
-            try {
-                writer.addDocument(facetsConfig.build(document));
-            } catch (IOException e) {
-                throw new IllegalArgumentException("Failed to create index");
+            if (CollectionUtils.isEmpty(context.getSampleNames())) {
+                VcfIndexEntry indexEntry = build(masterEntry, geneFiles, chromosome);
+                Document document = creator.buildDocument(indexEntry, vcfFile.getId());
+                try {
+                    writer.addDocument(facetsConfig.build(document));
+                } catch (IOException e) {
+                    throw new IllegalArgumentException("Failed to create index");
+                }
+            } else {
+                for (String sampleName: context.getSampleNames()) {
+                    if (!isEmptyStrain(context.getGenotype(sampleName))) {
+                        Set<String> sampleNames = new HashSet<>();
```

Comment:
We don't need to create a separate document for each sample, just one document with set of samples attached